numpy/lib/function_base.py

```diff
@@ -898,15 +898,14 @@ def average(a, axis=None, weights=None, returned=False):
     TypeError: Axis must be specified when shapes of a and weights differ.
 
     """
-    if not isinstance(a, np.matrix):
-        a = np.asarray(a)
+    a = np.asanyarray(a)
 
     if weights is None:
         avg = a.mean(axis)
         scl = avg.dtype.type(a.size/avg.size)
     else:
         a = a + 0.0
```

Comment:
What I meant was, that you can just add a 0. to the line below: `scl = wgt.sum(axis=axis, dtype=np.result_type(a.dtype, wgt.dtype, 0.))` to the same effect probably